### PR TITLE
scripts: extract_dts_includes: Generate defines for chosen props

### PR DESCRIPTION
**Pull request by galak**
Generate a set of defines that convey if a given chosen property is used
or not.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

### CODE REVIEW

**Comment by carlescufi** [comment]
This is in addition to what's being generated today right? i.e. `*_ON_DEV_NAME` and company?

**Comment by erwango** [comment]
Technically, change seem fine but as @carlescufi , I'm not sure how this should be used.
What will be the relationship with `CONFIG_UART_CONSOLE_ON_DEV_NAME` and `DT_UART_CONSOLE_ON_DEV_NAME` ?

Then, it brings me that, along with the addition of device bus info define, we need some way to document these kind of artifacts, as they are not documented through dt bindings..

**Comment by carlescufi** [comment]
@erwango they are now described here:
https://docs.zephyrproject.org/latest/devices/dts/device_tree.html#aliases-and-chosen-nodes

**Comment by erwango** [comment]
@carlescufi , ok, thanks. Though, my question was more related to the following symbols:
#DT_UART_CONSOLE_ON_DEV_NAME 1
(and the likes #DT_SENSOR_BUS_(SPI/I2C) 1

**Comment by galak** [comment]
> To address my point, I propose the following in the doc chapter provided by @carlescufi (`doc/devices/dts/device_tree.rst:311`):
> `CONFIG_*_ON_DEV_NAME` symbols could be used under `#ifdef DT_*_ON_DEV_NAME`.
> 
> Or maybe this is too much restrictive compare to current situation.

I think a general statement like:

As chosen properties tend to be related to software configuration, it can be useful for the build system to know if a chosen property was defined.  We generate a define for each chosen property, ie: 

`zephyr,flash` will generate: `#define DT_CHOSEN_ZEPHYR_FLASH 1`

File: doc/devices/dts/device_tree.rst [hunk 1]
```diff
@@ -308,6 +308,11 @@ The full set of Zephyr-specific ``chosen`` nodes follows:
    * - ``zephyr,uart-mcumgr``
      - :option:`CONFIG_UART_MCUMGR_ON_DEV_NAME`
 
+As chosen properties tend to be related to software configuration, it can be
+seful for the build system to know if a chosen property was defined. We
```
Comment (dbkinder) [inline doc/devices/dts/device_tree.rst]:
```suggestion
useful for the build system to know if a chosen property was defined. We
```

File: doc/devices/dts/device_tree.rst [hunk 2]
```diff
@@ -308,6 +308,11 @@ The full set of Zephyr-specific ``chosen`` nodes follows:
...
+generate a define for each chosen property.
```
Comment (dbkinder) [inline doc/devices/dts/device_tree.rst]:
```suggestion
generate a define for each chosen property, for example:
```

File: doc/devices/dts/device_tree.rst [hunk 3]
```diff
@@ -308,6 +308,11 @@ The full set of Zephyr-specific ``chosen`` nodes follows:
...
+
+``zephyr,flash`` will generate: `#define DT_CHOSEN_ZEPHYR_FLASH 1`
```
Comment (dbkinder) [inline doc/devices/dts/device_tree.rst]:
```suggestion
``zephyr,flash`` will generate: ``#define DT_CHOSEN_ZEPHYR_FLASH 1``
```

**Review by dbkinder — CHANGES_REQUESTED** [review]

**Comment by galak** [comment]
@dbkinder merged your changes in, take a look again.  Thanks.

**Review by dbkinder — APPROVED** [review]
LGTM, thanks!

**Review by carlescufi — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Pull request merged**